spec/models/validations/shared_validations_spec.rb

```diff
@@ -137,80 +141,96 @@
       end
 
       it "does not add an error if input is a multiple of ten" do
-        sales_log.savings = 30_000
+        sales_log.income1 = 30_000
         shared_validator.validate_numeric_step(sales_log)
         expect(sales_log.errors).to be_empty
       end
     end
 
     context "when validating a question with a step of 0.01" do
+      let(:step) { 0.01 }
+
       it "adds an error if input has more than 2 decimal places" do
-        sales_log.mscharge = 30.7418
+        sales_log.income1 = 30_123.7418
         shared_validator.validate_numeric_step(sales_log)
-        expect(sales_log.errors[:mscharge]).to include I18n.t("validations.shared.numeric.nearest_hundredth", field: "Monthly leasehold charges")
+        expect(sales_log.errors[:income1]).to include I18n.t("validations.shared.numeric.nearest_hundredth", field: "Buyer 1’s gross annual income")
       end
 
-      it "does not add an error if the user attempts to input a number in exponent format" do
-        sales_log.mscharge = "3e1"
+      it "does not add an error if input has 2 or fewer decimal places" do
+        sales_log.income1 = 30_123.74
         shared_validator.validate_numeric_step(sales_log)
         expect(sales_log.errors).to be_empty
       end
+    end
 
-      it "does not add an error if input has 2 or fewer decimal places" do
-        sales_log.mscharge = 30.74
+    context "when validating a question with a step of 0.1" do
+      let(:step) { 0.1 }
+
+      it "adds an error if input has more than 1 decimal place" do
+        sales_log.income1 = 30_123.74
+        shared_validator.validate_numeric_step(sales_log)
+        expect(sales_log.errors[:income1]).to include I18n.t("validations.shared.numeric.nearest_tenth", field: "Buyer 1’s gross annual income")
+      end
+
+      it "does not add an error if input has 1 or fewer decimal places" do
+        sales_log.income1 = 30_123.8
         shared_validator.validate_numeric_step(sales_log)
         expect(sales_log.errors).to be_empty
       end
     end
 
-    %i[sales_log lettings_log].each do |log_type|
-      describe "validate_owning_organisation_data_sharing_agremeent_signed" do
-        it "is valid if the Data Protection Confirmation is signed" do
-          log = build(log_type, :in_progress, owning_organisation: create(:organisation))
+    context "when validating a question with an unusual step" do
+      let(:step) { 0.001 }
 
-          expect(log).to be_valid
-        end
+      it "adds an appropriate error if input does not match" do
+        sales_log.income1 = 30_123.74
+        shared_validator.validate_numeric_step(sales_log)
+        expect(sales_log.errors[:income1]).to include I18n.t("validations.shared.numeric.nearest_step", field: "Buyer 1’s gross annual income", step: 0.001)
+      end
+    end
+  end
 
-        it "is valid when owning_organisation nil" do
-          log = build(log_type, owning_organisation: nil)
+  %i[sales_log lettings_log].each do |log_type|
```

Comment:
This section is just being pulled out of the describe block that it should never have been in - I've not made any changes.